src/ui/resource/Image.js

```diff
@@ -386,25 +392,37 @@ exports = Class(lib.PubSub, function () {
 		return _filterCanvas;
 	};
 
-	this._renderMultiply = function (ctx, srcX, srcY, srcW, srcH, color) {
-		// multiply rgb channels
-		var imgData = this.getImageData(srcX, srcY, srcW, srcH);
-		var data = imgData.data;
-		// simplified multiply math outside of the massive for loop
-		var a = color.a;
-		var mr = 1 + a * ((color.r / 255) - 1);
-		var mg = 1 + a * ((color.g / 255) - 1);
-		var mb = 1 + a * ((color.b / 255) - 1);
-		for (var i = 0, len = data.length; i < len; i += 4) {
-			data[i] *= mr;
-			data[i + 1] *= mg;
-			data[i + 2] *= mb;
+	var unusedCanvas = null;
+
+	this._getTintedImage = function(srcX, srcY, srcW, srcH, color) {
+		var colorString = (color.r << 16 | color.g << 8 | color.b).toString(16);
+		var cacheKey = this.getURL() + srcX + "|" + srcY + "|" + srcW + "|" + srcH + "|" + colorString;
+		var result = TintedImageCache.get(cacheKey);
+		if (!result) {
+			result = unusedCanvas || new Canvas();
+			resultCtx = result.getContext('2d');
+			result.width = srcW;
+			result.height = srcH;
+			var imgData = this.getImageData(srcX, srcY, srcW, srcH);
+			var data = imgData.data;
+			// simplified multiply math outside of the massive for loop
+			var a = color.a;
+			var mr = 1 + a * ((color.r / 255) - 1);
+			var mg = 1 + a * ((color.g / 255) - 1);
+			var mb = 1 + a * ((color.b / 255) - 1);
+			for (var i = 0, len = data.length; i < len; i += 4) {
+				data[i] *= mr;
+				data[i + 1] *= mg;
+				data[i + 2] *= mb;
+			}
+			// put the updated rgb data into our new canvas
+			// console.log("CACHING TINTED IMAGE", cacheKey, color.r, color.g, color.b);
+			resultCtx.putImageData(imgData, 0, 0);
+			var removedEntry = TintedImageCache.put(cacheKey, result);
+			unusedCanvas = removedEntry ? removedEntry.value : null;
+			console.log(unusedCanvas);
```

Comment:
Ah yeah, the console.log can go bye bye. The unusedCanvas is an optimization that reuses an existing canvas instead of constantly creating/destroying them, but this only comes into play once you hit the cache limit. An absolute requirement? Probably not, but doesn't hurt.